lib/rules/no-misleading-character-class.js

```diff
@@ -227,59 +228,61 @@ module.exports = {
         const parser = new RegExpParser();
 
         /**
-         * Generates a granular loc for context.report, if directly calculable.
-         * @param {Character[]} chars Individual characters being reported on.
+         * Creates a function used to generate locs for a specified node.
          * @param {Node} node Parent string node to report within.
-         * @returns {Object | null} Granular loc for context.report, if directly calculable.
-         * @see https://github.com/eslint/eslint/pull/17515
+         * @returns {(matches: Character[][]) => Location[]} A function to generate locs for the node.
          */
-        function generateReportLocation(chars, node) {
+        function createReportLocationGenerator(node) {
 
-            // Limit to to literals and expression-less templates with raw values === their value.
-            switch (node.type) {
-                case "TemplateLiteral":
-                    if (node.expressions.length || sourceCode.getText(node).slice(1, -1) !== node.quasis[0].value.cooked) {
-                        return null;
-                    }
-                    break;
-
-                case "Literal":
-                    if (typeof node.value === "string" && node.value !== node.raw.slice(1, -1)) {
-                        return null;
-                    }
-                    break;
-
-                default:
-                    return null;
+            // Only literals and expression-less templates generate granular errors.
+            if (!(node.type === "TemplateLiteral" && !node.expressions.length || node.type === "Literal")) {
```

Comment:
Funny, I was going to suggest `if (node.type !== "TemplateLiteral" || (node.type === "Literal" && node.expressions.length)) {`. No preference between the two from me. 🙂

Reply:
Thanks! I realized that that check was always falsy, so I just removed it. It's because strings and templates without expressions always produce granular reports now, and so the default case isn't hit any more. Somehow I missed that while testing.